### DISCUSSION TITLE
Validate correctness of MDAnalysis selection command without the universe present

### DISCUSSION THREAD

**Discussion by marinegor** [MDAnalysis/mdanalysis#4767] [General]
I'm looking into a function like `validate` that does roughly this:

```python
import MDAnalysis as mda

valid_selection = 'name N and not protein'
typo_in_selection = 'namme N and not protein'
empty_selection = 'name N and name O'

mda.??.validate(valid_selection)   # OK
mda.??.validate(typo_in_selection) # Exception
mda.??.validate(empty_selection)   # Warning: selection is guaranteed to be empty
```

The workaround I can come up with is to create a dummy universe with some atoms inside and then try selecting atoms from it, but perhaps there's a way to do that in a less hacky way?

**Comment by orbeckst** [comment]
The typo check should really just be a sensible exception from the selection string parser. 

One of the longest-standing open issues is the suggestion to use *pyparsing* instead of our own handcrafted selection parser https://github.com/MDAnalysis/mdanalysis/issues/371 ; this change may come with more robust error recognition and handling (and better performance and probably also more customizability).
